File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/WindowElements/getwindowelements.rb
module Asposediagramjava
  module GetWindowElements
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        windows = diagram.getWindows()

        i = 0
        while i < windows.getCount()
            window = windows.get(i)
            puts "ID: " + window.getID().to_s
            puts "Type: " + window.getWindowType().to_s
            puts "Window height: " + window.getWindowHeight().to_s
            puts "Window width: " + window.getWindowWidth().to_s
            puts "Window state: " + window.getWindowState().to_s

            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithWindowElements/GetWindowElements.php
<?php
namespace Aspose\Diagram\WorkingwithWindowElements;

use com\aspose\diagram\Diagram as Diagram;
use com\aspose\diagram\SaveFileFormat as SaveFileFormat;
class GetWindowElements{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir."Drawing.vsd");

        $windows=$diagram->getWindows();

        $i = 0;
        while ($i<(int)(string)$windows->getCount()) {
            $window=$windows->get($i);
            print "ID: ".(string)$window->getID();
            print "Type: ".(string)$window->getWindowType();
            print "Window height: ".(string)$window->getWindowHeight();
            print "Window width: ".(string)$window->getWindowWidth();
            print"Window state: ".(string)$window->getWindowState();
            $i+= 1;
        }
    }
}